colorcalculator.php

```diff
@@ -0,0 +1,74 @@
+<html>
+	<body>
+		<h1> Color Calculator </h1> <br> <br>
+			<form action=""
+
+				<p>First Color
+						<select name="color1" class="color1">
+							<option value='0'>--Select function--</option>
+							<option value='Red'> Red </option>
+							<option value='Yellow'> Yellow </option>
+							<option value='Blue'> Blue </option>
+						</select>
+				</p>
+				<br>
+				<p> mixed with </p>
+				<br>
+				<p>Second Color
+						<select name="color2" class="color2">
+							<option value='0'>--Select function--</option>
+							<option value='Red'> Red </option>
+							<option value='Yellow'> Yellow </option>
+							<option value='Blue'> Blue </option>
+						</select>
+				</p>
+				<input type= "submit"/> <br>
+
+			</form>
+	</body>
+</html>
+
+<?php
+
+//all error messages
+if (isset($_REQUEST['color1']) && isset($_REQUEST['color2']) && $_REQUEST['color1']=='0') {
+	echo "<h1>Please choose two colors.</h1>";
+} elseif (isset($_REQUEST['color1']) && isset($_REQUEST['color2']) && $_REQUEST['color2']=='0') {
+	echo "<h1>Please choose two colors.</h1>";
+} elseif (isset($_REQUEST['color1']) && isset($_REQUEST['color2']) && $_REQUEST['color2']=='0' && $_REQUEST['color1']=='0') {
+	echo "<h1>Please choose two colors.</h1>";
+// all the possible colors
+} elseif (isset($_REQUEST['color1']) && isset($_REQUEST['color2']) && $_REQUEST['color1']=='Red' && $_REQUEST['color2']=='Blue') {
```

Comment:
You don't really need to check if colors 1 and 2 are set in each one of these. Because of how an if/elseif works, if the initial `if` statement is true, it'll never consider running the later ones. So if you take my suggestion above to check that the colors are set at the beginning, you can assume they're set here or else it'd never get to this section of code.